targets/HUDL/main.cpp

```diff
@@ -13,53 +20,180 @@
+    auto hudl = HUDL::HUDL(regSelect, reset, hudl_spi);
 
-    // String to store user input
-    char buf[100];
+    // Reserved memory for CANopen stack usage
+    uint8_t sdoBuffer[1][CO_SDO_BUF_BYTE];
+    CO_TMR_MEM appTmrMem[4];
 
-    while (1) {
-        // Read user input
-        uart.printf("Enter message: ");
-        uart.gets(buf, 100);
+    // Attempt to join the CAN network
+    IO::CAN::CANStatus result = can.connect();
 
-        // Initialized LCD
-        board.initLCD();
+    //test that the board is connected to the can network
+    if (result != IO::CAN::CANStatus::OK) {
+        uart.printf("Failed to connect to CAN network\r\n");
+        return 1;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Setup CAN configuration, this handles making drivers, applying settings.
+    // And generally creating the CANopen stack node which is the interface
+    // between the application (the code we write) and the physical CAN network
+    ///////////////////////////////////////////////////////////////////////////
+    // Make drivers
+    CO_IF_DRV canStackDriver;
+
+    CO_IF_CAN_DRV canDriver;
+    CO_IF_TIMER_DRV timerDriver;
+    CO_IF_NVM_DRV nvmDriver;
+
+    IO::getCANopenCANDriver(&can, &canOpenQueue, &canDriver);
+    IO::getCANopenTimerDriver(&timer, &timerDriver);
+    IO::getCANopenNVMDriver(&nvmDriver);
+
+    canStackDriver.Can = &canDriver;
+    canStackDriver.Timer = &timerDriver;
+    canStackDriver.Nvm = &nvmDriver;
 
-        // Clear LCD Screen
-        // clear_lcd();
+    //setup CANopen Node
+    CO_NODE_SPEC canSpec = {
+        .NodeId = HUDL::HUDL::NODE_ID,
+        .Baudrate = IO::CAN::DEFAULT_BAUD,
+        .Dict = hudl.getObjectDictionary(),
+        .DictLen = hudl.getObjectDictionarySize(),
+        .EmcyCode = NULL,
+        .TmrMem = appTmrMem,
+        .TmrNum = 16,
+        .TmrFreq = 100,
+        .Drv = &canStackDriver,
+        .SdoBuf = reinterpret_cast<uint8_t*>(&sdoBuffer[0]),
+    };
 
-        board.drivePixel(1, 1, 1, 255);
-        time::wait(10000);
+    CO_NODE canNode;
+
+    CONodeInit(&canNode, &canSpec);
+    CONodeStart(&canNode);
+    CONmtSetMode(&canNode.Nmt, CO_OPERATIONAL);
+
+    time::wait(500);
+
+    //print any CANopen errors
+    uart.printf("Error: %d\r\n", CONodeGetErr(&canNode));
+    while (1) {
+        //Print new value when changed over CAN
+        uint32_t* temps = hudl.getThermTemps();
+        // Process incoming CAN messages
+        for (int i = 0; i < 4; i++) {
+            uart.printf("Temp %d: %d\n\r", i, *(temps + i));
+        }
 
-        // echoes command back
-        uart.printf("\n\recho: %s\n\r", buf);
+        CONodeProcess(&canNode);
+        // Update the state of timer based events
+        COTmrService(&canNode.Tmr);
+        // Handle executing timer events that have elapsed
+        COTmrProcess(&canNode.Tmr);
+        // Wait for new data to come in
+        time::wait(1000);
```

Comment:
I believe for most boards this is 100 ms. Is there a reason you lengthened it here?